### PR TITLE
Slurm tests. Wait for all partitions to became "UP"

### PR DESCRIPTION
**Pull request by mr0re1**
### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml [hunk 1]
```diff
@@ -18,7 +18,15 @@
     that:
     - custom_vars.partitions is defined
     - custom_vars.mounts is defined
-
+- name: Ensure all partitions are UP
+  ansible.builtin.shell: sinfo --noheader  --all --format "%P[%a]" | grep -v "\[up\]" | cat
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml:22]: It's to swallow error code by `grep` that fires if no lines were selected.

[the remark this answers is not in the document]
Comment (tpdownes) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml:22]: Please add a comment about that and then merge.

The more "ansible-native" way would be to set `failed_when: false` or `failed_when: partitions_down.rc == 2` or similar. You could also do something thing like `until: partitions_down.rc == 1`. A comment is sufficient though as we need this fix.

**Review by tpdownes — APPROVED** [review]

File: tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mr0re1) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml:28]: Added a comment \w TODO

**Pull request merged**